src/python/zquantum/core/measurement.py

```diff
@@ -412,48 +428,69 @@ def convert_bitstring_to_int(bitstring: Sequence[int]) -> int:
     return int("".join(str(bit) for bit in bitstring[::-1]), 2)
 
 
-def check_parity(
-    bitstring: Union[str, Sequence[int]], marked_qubits: Iterable[int]
-) -> bool:
-    """Determine if the marked qubits have even parity for the given bitstring.
+def check_parity_of_vector(
```

Comment:
[request] You shouldn't replace `check_parity` method with some other, as it breaks API – libraries depending on `z-quantum-core` might be using it. I would just add this as a new method.

Reply:
Also, it is called `check_parity_of_vector`, but actually doesn't really take in vector – (at least I wouldn't call "list of strings" a vector).
I think in the case of this method, since speed is important, it's better to just say that it takes only one type of inputs (probably `np.ndarray`) which is efficient.